src/nvim/globals.h

```diff
@@ -838,13 +834,6 @@ EXTERN int ctrl_x_mode INIT(= 0);       /* Which Ctrl-X mode are we in? */
 
 EXTERN int no_abbr INIT(= TRUE);        /* TRUE when no abbreviations loaded */
 
-#ifdef USE_EXE_NAME
-EXTERN char_u   *exe_name;              /* the name of the executable */
```

Comment:
`:help $VIM`, item 3, reveals that this exists to support running vim from any old directory, and it walks up from the exe location to look for a user `.vimrc` file.

Regardless if we want to support this, the platform-specific nature of this define is unnecessary. We can easily add support for this feature later if needed.